File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> answer = new ArrayList<>();
        int[] days = new int[progresses.length];
        
        for (int i = 0; i < progresses.length; i++) {
            days[i] = (100 - progresses[i] + speeds[i] - 1) / speeds[i]; 
        }
        
        int day = days[0];
        int cnt = 1;
        
        for(int i = 1; i < days.length; i++) {
            if(days[i] <= day)  cnt++;
            else {
                answer.add(cnt);
                day = days[i];
                cnt = 1;
            }
        }
        
        answer.add(cnt);
        
        return answer.stream()
            .mapToInt(i -> i)
            .toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    vector<int> worked;
    for(int i = 0; i < speeds.size(); i++) {
        int progress = progresses[i], speed = speeds[i];
        
        int finishedPeriod = (100 - progresses[i]) / speed;
        if((100 - progress) % speed != 0) finishedPeriod++;
        
        worked.push_back(finishedPeriod);
    }
    
    int cnt = 1;
    int maxDay = worked[0];
    
    for(int i = 1; i < worked.size(); i++) {
        int nowPeriod = worked[i];
        
        if (maxDay >= nowPeriod) cnt++;
        else {
            maxDay = nowPeriod;
            answer.push_back(cnt);
            cnt = 1;
        }
    }
    
    answer.push_back(cnt);
    
    return answer;
}